apps/namadillo/src/utils/index.ts

```diff
@@ -122,12 +122,14 @@ const findDisplayUnit = (asset: Asset): AssetDenomUnit | undefined => {
 };
 
 const isNamAsset = (asset: Asset): boolean =>
-  asset.symbol === namadaAsset.symbol;
+  asset.symbol === namadaAsset().symbol;
 
 export const toDisplayAmount = (
   asset: Asset,
   baseAmount: BigNumber
 ): BigNumber => {
+  if (isNamAsset(asset)) return baseAmount;
```

Comment:
I think this won't work for NAM transferred from other chains. The indexer will return the balances in display denom for NAM from the current chain, but in base denom for NAM from other chains. But this check here will assume any token with `symbol` as `NAM` is in display denom.

I was wondering if this check should be outside the `toDisplayAmount` function anyway, since to me it seems like that logic is outside of the scope of the function. What do you think?

Reply:
Actually, I just also realised I think it's not going to work for displaying NAM on Cosmos chains either. The NAM balance we get from a Cosmos chain will be in base denom, but this check will stop it being converted to display denom.

Reply:
yeah maybe move this outside makes sense :D Just so I understand better, what do you mean by "NAM  transferred from other chains." For NAM to arrive on other chain it first has to be sent from namada chain. So if we send NAM to cosmos and we send it back, will be it treated as different token than regular NAM?

Reply:
The two cases I think won't work are:

1. NAM that has been sent to a Cosmos chain (since the balance query for cosmjs returns amounts in base denom)
2. NAM from something other than the current chain e.g. your current chain is housefire but you sent some NAM from internal devnet to housefire

In both cases, these should display properly as the Namada asset in Namadillo, but the denomation will be wrong in this PR because Namadillo thinks it's already in display denom but it's actually in base denom.

The case you're talking about where you send NAM from Namada to Cosmos and then back should work fine I think, because the indexer will still return the balance using the original tnam address and in display denom (as far as I know).